node_helper.js

```diff
@@ -570,23 +574,42 @@ module.exports = NodeHelper.create({
     for (var id of objectIDs) {
       var url = `https://collectionapi.metmuseum.org/public/collection/v1/objects/${id}`;
 
-      request(url, function (error, response, body) {
-        var obj = JSON.parse(body);
+      const req = https.request(url, (res) => {
```

Comment:
Bleh, plain `request` calls should be replaced with `fetch` calls instead.  Something like:

```js
fetch(url)
  .then(response => response.json())
  .then(obj => {
    if (obj.isPublicDomain) {
    ...
  })
  .finally(() => {
    if (--pendingRequests == 0) {
      self.cacheResult(...);
    }
  });
```

Can you try it that way and see if it works?

Reply:
That should do the trick as well :)

Also we have to either fix docs or the approach on feeding multiple department ids - as unfortunately the API doesn't support multiple IDs :disappointed: